### DISCUSSION TITLE
How to test if spell checker is working?

### DISCUSSION THREAD

**Discussion by Gitoffthelawn** [florisboard/florisboard#1310] [Q&A]
I installed Mozilla's Firefox spell checker database into FlorisBoard without error.  But so far, I haven't seen any difference in functionality.  What is a good test to see if it is working?  I tried just going into an SMS app and a text editor and typing nonsense, but no spelling errors were detected.

**Comment by patrickgold** [comment]
If you are in the beta spell checking bugs out for language codes other than `en[*]` sometimes but I am currently working on a new spelling manager and fixing this issue with that.

Generally the best apps to test them out are SMS/messaging apps, if no red lines appear in the text box for nonsense then something is not working. Also make sure that you set the spell checking language to system default (then only takes the language of your system) or to follow keyboard subtype language in the system UI for spell checking.

**Reply by Gitoffthelawn** [reply]
Thanks Patrick.

Setting everything to system default didn't work, but setting up a keyboard subtype language made the spell-checker work right away.

BTW, I used FlorisBoard for a very short time before adding the spell-checker.  Before I added the spell-checker, FlorisBoard never crashed.  After I added the spell-checker, FlorisBoard now crashes somewhat regularly.  But as I mentioned, I didn't have FlorisBoard installed for very long before adding the spell-checker, so it might not be causal.  Do you think the spell-checker is likely increasing the number of crashes, or that the crashes are likely not caused by using the spell-checker?

The main crash I experience is when I unlock the device, load an SMS app, and start creating a new message.  After typing 2-3 characters, FlorisBoard usually crashes.  If I then re-open the SMS app, things typically work fine until the device locks again.  Then I have the same experience.

Overall, I look forward to FlorisBoard getting better and better over time. 👍🏾

**Reply by patrickgold** [reply]
If FlorisBoard often crashes, first check if you have word suggestions enabled - this is the current no. 1 source of crash reports. The spell checker is typically a very stable thing, although I am currently integrating it in the extension foundation, so I can't test it out in the latest debug build.

> The main crash I experience is when I unlock the device, load an SMS app, and start creating a new message. After typing 2-3 characters, FlorisBoard usually crashes.

It would be interesting to know what the crash log is. If you experience this crash the next time, make sure to copy/paste it into this thread so I can see what exactly is crashing!

> Overall, I look forward to FlorisBoard getting better and better over time. 👍🏾

I am trying my best - things will definitely improve and I have a lot planned, it will just take a bit of time sadly because my free time isn't unlimited.

**Reply by Gitoffthelawn** [reply]
> If FlorisBoard often crashes, first check if you have word suggestions enabled - this is the current no. 1 source of crash reports. The spell checker is typically a very stable thing, although I am currently integrating it in the extension foundation, so I can't test it out in the latest debug build.

Yes, I do have them enabled.  When I first installed FlorisBoard I was a little bit dismayed that it didn't seem to have word suggestions.  I find word suggestions to be *very* helpful when they are of a good quality.  When I discovered that FlorisBoard had a word suggestion option, I enabled it right away.  I agree that I think that functionality is likely causing the crashes.

> It would be interesting to know what the crash log is. If you experience this crash the next time, make sure to copy/paste it into this thread so I can see what exactly is crashing!

I'll get it for you.  IIRC, it was an OOM error, but I'll get more details for you when I have a little spare time.  My recollection of OOM errors matches my recollection of something you wrote about the word suggestion feature causing OOM errors.  So those recollections support each other. :)

I had to switch away from FlorisBoard for now because, although I like it for *many* reasons more than any other keyboard, the crashes were really getting in the way.  But I expect to return to FlorisBoard once it's matured just a little more and you've had enough time to work out some of the big kinks.

> I am trying my best - things will definitely improve and I have a lot planned, it will just take a bit of time sadly because my free time isn't unlimited.

Yep.  I understand and relate completely.  I think you've done an amazing job so far.  I have quite a few ideas to contribute regarding functionality that can make FlorisBoard unique and a better productivity-enhancer than any other keyboard.  I'm hoping to have some free time (at some point) to write up those ideas.  Unfortunately, like you, free time is very limited right now.  I'm actually just taking a quick breather from obligations to write this to you now... ⌨️ 

You mentioning that you are planning on continuing your efforts encourages me to try to find some time to write up at least some of those ideas.  I feel like the project right now is at the tipping point where other projects either get abandoned or improve greatly.

Take care, and talk with you later.
